paywall/src/data-iframe/postOfficeListener.js

```diff
@@ -64,6 +66,21 @@ export const makePurchaseKeyListener = (logger, purchase) =>
     purchase(lock, extraTip)
   }
 
+export const makeWalletInfoListener = window =>
+  async function walletInfoListener(walletInfo) {
+    if (
+      !walletInfo ||
+      typeof walletInfo !== 'object' ||
+      Array.isArray(walletInfo)
+    ) {
+      return
+    }
+    if (walletInfo.noWallet) {
```

Comment:
I think we might be able to update the conditional here to be something like `walletInfo && walletInfo.noWallet` otherwise return?

Reply:
I considered this. There is a chance we may want to act upon `walletInfo.isEnabled` as well (i.e. user has metamask, but has chosen not to enable it for this page, we should probably reset their account, even though metamask shows their account when calling `eth_listAccounts`). In that case, the line would become `walletInfo.noWallet || !walletInfo.isEnabled`. Would be interested in your take on this?